### ISSUE TITLE
Plugins don't formally depend on any package that contains their informal type dependencies

### ISSUE THREAD

**Issue by garybernhardt** [bugsnag/bugsnag-js#997]
**Background**:

* The plugin-express module (and other plugins, I assume) depends on bugsnag/core, but doesn't depend on bugsnag-js.
* The type definitions for Bugsnag are in bugsnag-js (not bugsnag/core), so the plugins don't formally depend on the package that contains the types.
* But the plugins *do* informally depend on the types, for obvious reasons: if the types don't match, then code using the plugins won't build.

**Bug**:

This caused a routine automated depfu upgrade to break for us: it upgraded plugin-express to 7.3.0, but left bugsnag-js at 7.2.1. That's allowed given the npm modules' declared dependencies. But that combination fails to build because any code that attempts to add a 7.3.0 plugin to a 7.2.1 bugsnag instance gets a type error.

**Reproduction steps**:

Install plugin-express 7.3.0 and bugsnag-js 7.2.1. Then try to build this TypeScript code:

```typescript
import Bugsnag from '@bugsnag/js'
import bugsnagExpress from "@bugsnag/plugin-express"

Bugsnag.start({
  apiKey: 'MY_API_KEY',
  plugins: [
    bugsnagExpress
  ],
})
```

**Comment by phillipsam** [comment]
Hi @garybernhardt thanks for the report! We will look into getting this fixed.

**Comment by phillipsam** [comment]
We have now released a fix for this in [v7.3.3](https://github.com/bugsnag/bugsnag-js/releases/tag/v7.3.3).

**Comment by garybernhardt** [comment]
Thanks @phillipsam!
